### PR TITLE
Improve exception handling for `CREATE TABLE` task

### LINKED ISSUE #10446: Misleading exception when trying to create Iceberg table if Hive table with same name already exists
```
create table hive.default.commontable (id bigint);
create table iceberg.default.commontable (id bigint);
```

The second query fails with the exception:

```
[2022-01-03 06:43:31] [84148224] Query failed (#20220103_054331_00040_cc26e): Not an Iceberg table: default.commontable
```

This message is rather misleading and should be changed with a meaningful message for the end user pointing out that a table with the same name and schema already exists in the hive metastore.

The issue occurs in `io.trino.plugin.iceberg.catalog.AbstractMetastoreTableOperations#refresh` method 

```
 if (!isIcebergTable(table)) {
            throw new UnknownTableTypeException(getSchemaTableName());
}
```

However, due to the fact that the connectors `hive` and `iceberg` share the same hive metastore it is rather difficult to give a meaningful message in this case.

Executing the following statements

```
create table hive.default.commontable (id bigint);
create table hive.default.commontable (id bigint);
```

leads to the following error (as expected): 

```
Query failed (#20220103_143437_00019_y848r): line 1:1: Table 'hive.default.commontable' already exists
```

Related issues:

#8693

### PR DESCRIPTION
**Pull request by findinpath**
Provide meaningful exception message for `CREATE TABLE` task
when dealing with namesake table of unsupported type in
shared metastore.

This change affects the connectors: hive, iceberg, delta lake
which may deal with a shared metastore (HMS/AWS Glue).

## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

This change affects the core query engine (`CreateTableTask`)

> How would you describe this change to a non-technical end user or system administrator?

In the context of using shared metastore between Hive, Iceberg/Delta it may very well happen that the users are trying to create a namesake table from two different connectors. This PR improves exception handling in the `CREATE TABLE` task to provide a meaningful message to the user in such a situation.

## Related issues, pull requests, and links

Fixes #10446

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# Main
* Improve exception handling in`CREATE TABLE` task when dealing with namesake table of unsupported type in
shared metastore.
```

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/StandardErrorCode.java [hunk 1]
```diff
@@ -129,6 +129,7 @@
     MISSING_ROW_PATTERN(106, USER_ERROR),
     INVALID_WINDOW_MEASURE(107, USER_ERROR),
     STACK_OVERFLOW(108, USER_ERROR),
+    UNSUPPORTED_TABLE_TYPE(109, USER_ERROR),
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/StandardErrorCode.java:132]: cc @electrum

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -560,7 +561,7 @@ private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, Sche
                 .orElseThrow(() -> new TableNotFoundException(tableName));
 
         if (isIcebergTable(table) || isDeltaLakeTable(table)) {
-            throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, format("Not a Hive table '%s'", tableName));
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:563]: should we remove HIVE_UNSUPPORTED_FORMAT  now?

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergErrorCode.java [hunk 3]
```diff
@@ -24,7 +24,7 @@
 public enum IcebergErrorCode
         implements ErrorCodeSupplier
 {
-    ICEBERG_UNKNOWN_TABLE_TYPE(0, EXTERNAL),
+    // code ICEBERG_UNKNOWN_TABLE_TYPE(0, EXTERNAL) is deprecated
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergErrorCode.java:27]: simply remove?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveTablesCompatibility.java [hunk 4]
```diff
@@ -65,4 +65,28 @@ public void testHiveSelectFromIcebergTable()
 
         onTrino().executeQuery("DROP TABLE iceberg.default." + tableName);
     }
+
+    @Test(groups = {ICEBERG, STORAGE_FORMATS, HMS_ONLY})
+    public void testIcebergCannotCreateTableNamesakeToHiveTable()
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveTablesCompatibility.java:70]: `testIcebergCreateTableWhenHiveTableExists`

Comment (findinpath) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveTablesCompatibility.java:70]: Changed to `testIcebergCreateTableWhenNamesakeHiveTableExists`

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergErrorCode.java (hunk 3 above)
Comment (findinpath) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergErrorCode.java:27]: I'm following the pattern `code XXX(YY) is deprecated` which is used in other `...ErrorCode` classes.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 2 above)
Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:563]: The error code still has about `10` usages and is used to denote compression codec problems , storage format problems.

**Review by alexjo2144 — APPROVED** [review]
Looks reasonable to me.

There's a bigger question along with https://github.com/trinodb/trino/pull/11617 I guess. Should connectors ever hide unsupported tables? Now that we have table redirections it seems to me like we should never filter tables, and so these create table statements should always see the existing tables rather than relying on catching it with the exception code.

**Comment by findinpath** [comment]
@alexjo2144  good point.

I would rather put this PR on hold until #11617  is cleared and then act accordingly.

Calling `io.trino.metadata.Metadata#listTables` instead of `io.trino.metadata.Metadata#getTableHandle(io.trino.Session, io.trino.metadata.QualifiedObjectName)` would spare us of the exception code matching strategy which is relatively error prone.

**Review by findepi — APPROVED** [review]

**Comment by mosabua** [comment]
I will assume that the label "no-release-notes" from @findepi supersedes the suggestion from @findinpath

**Pull request merged**